src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -162,33 +350,60 @@ public async Task ProcessRequestAsync<TContext>(IHttpApplication<TContext> appli
                     }
                 }
             }
+            catch (Http3StreamErrorException ex)
+            {
+                error = ex;
+                //errorCode = ex.ErrorCode;
+                Abort(new ConnectionAbortedException(ex.Message, ex), ex.ErrorCode);
+            }
             catch (Exception ex)
             {
                 error = ex;
-                Log.LogWarning(0, ex, "Stream threw an exception.");
+                Log.LogWarning(0, ex, "Stream threw an unexpected exception.");
             }
             finally
             {
                 var streamError = error as ConnectionAbortedException
                     ?? new ConnectionAbortedException("The stream has completed.", error);
+
+                // Input has completed.
+
+                Input.Complete();
+                _context.Transport.Input.CancelPendingRead();
```

Comment:
`Input` is `_context.Transport.Input`. Why are we calling CancelPendingRead after we're done reading?